webdriverio/web/examples/typescript/test/pageobjects/login.page.ts

```diff
@@ -0,0 +1,23 @@
+import Page from './page';
+
+/**
+ * sub page containing specific selectors and methods for a specific page
+ */
+export class LoginPage extends Page {
+    /**
+     * a method to encapsule automation code to interact with the page
+     * e.g. to login using username and password
+     */
+    login (username: string, password: string) {
+        $('#username').setValue(username);
+        $('#password').setValue(password);
+        $('button[type="submit"]').click(); 
+    }
+
+    /**
+     * overwrite specifc options to adapt it to page object
+     */
+    open () {
+        return super.open('login');
+    }
+}
```

Comment:
This is how you could write the PO with all proposed feedback

```ts
import Page from './page';

/**
 * sub page containing specific selectors and methods for a specific page
 */
class LoginPage extends Page {
    // Methods are made private because we don't want to expose 
    // selectors to the specs
    private get username ():WebdriverIO.Element{
        return $('#username')
    }
    
    private get password ():WebdriverIO.Element{
        return $('#password')
    }
    
    private get submit ():WebdriverIO.Element{
        return $('button[type="submit"]')
    }

    /**
     * a method to encapsule automation code to interact with the page
     * e.g. to login using username and password
     */
    login (username: string, password: string):void {
        this.username.setValue(username);
        this.password.setValue(password);
        this.submit.click();
    }

    /**
     * overwrite specifc options to adapt it to page object
     */
    open ():void {
        return super.open('login');
    }
}

export default new LoginPage();
```